src/pages/Account/Tabs/ModulesTab/Contract.tsx

```diff
@@ -299,23 +300,24 @@ function RunContractForm({
 
   const onSubmit: SubmitHandler<ContractFormType> = async (data) => {
     logEvent("write_button_clicked", fn.name);
-    const payload: Types.TransactionPayload = {
-      type: "entry_function_payload",
-      function: `${module.address}::${module.name}::${fn.name}`,
-      type_arguments: data.typeArgs,
-      arguments: data.args.map((arg, i) => {
-        const type = fnParams[i];
-        if (type.includes("vector")) {
-          // when it's a vector<u8>, we support both hex and javascript array format
-          return type === "vector<u8>" && arg.trim().startsWith("0x")
-            ? Array.from(new HexString(arg).toUint8Array()).map((x) =>
-                x.toString(),
-              )
-            : deserializeVector(arg);
-        } else if (type.startsWith("0x1::option::Option")) {
-          arg ? {vec: [arg]} : undefined;
-        } else return arg;
-      }),
+    const payload: InputTransactionData = {
+      data: {
+        function: `${module.address}::${module.name}::${fn.name}`,
+        typeArguments: data.typeArgs,
+        functionArguments: data.args.map((arg, i) => {
+          const type = fnParams[i];
+          if (type.includes("vector")) {
+            // when it's a vector<u8>, we support both hex and javascript array format
+            return type === "vector<u8>" && arg.trim().startsWith("0x")
+              ? Array.from(new HexString(arg).toUint8Array()).map((x) =>
+                  x.toString(),
+                )
+              : deserializeVector(arg);
+          } else if (type.startsWith("0x1::option::Option")) {
+            arg ? {vec: [arg]} : undefined;
+          } else return arg;
+        }),
+      },
```

Comment:
This unfortunately needs to be reworked, so upgrading to SDK v2 is blocked on this.

SDK V2 in its argument parsing doesn't accept strings for every input, so they'd first have to be parsed into their respective types before passing them on.